src/effect.cpp

```diff
@@ -8,10 +8,32 @@
 #include <map>
 #include <sstream>
 
-std::map<std::string, effect_type> effect_types;
+namespace {
+std::map<efftype_id, effect_type> effect_types;
+}
+
+template<>
+const effect_type& string_id<effect_type>::obj() const
+{
+    const auto iter = effect_types.find( *this );
+    if( iter == effect_types.end() ) {
+        debugmsg( "invalid effect type id %s", c_str() );
+        const effect_type dummy{};
```

Comment:
This broke Jenkins:

src/effect.cpp: In member function 'const T& string_id<T>::obj() const [with T = effect_type]':
src/effect.cpp:21:27: error: reference to local variable 'dummy' returned [-Werror=return-local-addr]
         const effect_type dummy{};
                           ^
cc1plus: all warnings being treated as errors
make: *** [objwin/tiles/effect.o] Error 1
make: *** Waiting for unfinished jobs....
Build step 'Execute shell' marked build as failure